### ISSUE TITLE
"today" remains on yesterday's date

### ISSUE THREAD

**Issue by marc-medley** [nutritionfactsorg/daily-dozen-ios#42]
**Overview**

If the app is left overnight as the last app used , then "today" may remain on yesterday's date without a clear way to go to today's actual date.

**Use Case**

When a user checks the Daily Dozen as the end of the day, it would be useful to be able to navigate to the current date first thing in the morning.

**Configuration**

The modality of the restriction varies by the OS version.

* Daily Dozen 3.2.4 (current main branch)

* iOS 12.4: date tumbler rolls back to yesterday's date 

![image](https://user-images.githubusercontent.com/15098151/98909576-dac4bf00-2476-11eb-818e-4c37b121ce47.png)

* iOS 14.2: date picker popup selector has today's grayed out and the "Back to today" is not accessible.

![image](https://user-images.githubusercontent.com/15098151/98909738-12336b80-2477-11eb-8351-c256163f5099.png)

**Steps To Reproduce**

_iPhone, personal_

1. Open Daily Dozen checklist as the last activity of the phone for the day.
2. Touch "today" so that the date is showing. Set the phone aside.
3. On the next day, go directly back to the Daily Dozen checklist and try to navigate to the actual current day. 

_iPhone, test unit_

> Note: a test phone is preferred when changing the device date, since a device level date change can affect other files on the phone.

1. Open Daily Dozen checklist to today. 
2. Touch "today" so that the date is showing.
3. Go home. Then to go `Settings`.
4. In `Settings`, go to the General > Data & Time Screen
    * Disable "Set Automatically" to turn off network time
    * Change the current date
5. Return to the Daily Dozen app checklist. Try to navigate to the just set date.

_Simulator_

The iOS simulator does not have a mechanism to switch the simulator device date when the simulator and application are running.

**Related Issues**

Zen Desk 36332

**Possible Fix**

TBD.

**Comment by marc-medley** [comment]
`dateButton`, `dateButtonPressed`

* toggle back to display "Today" text if current display date equals actual today date 

`datePicker`

* needs to be able to reach current date after an overnight device date change

`backButton`, `backButtonPressed`

* "Back to today" needs to go to a new blank list after an overnight device date change
    * testing needs to verify that no current view data is lost

`viewSwiped`

* "Back to today" should go to a new blank "Today" screen after an overnight device date change 

_Item History_

* Calendar view should include current date as selectable after an overnight device date change

**Comment by marc-medley** [comment]
`Date()` as been wrapped in `DateManager` to allow for testing of overnight and timezone date change scenarios.

``` swift
struct DateManager {
    
    static private var _currentDatetime = Date()
    
    /// return "now" date and time
    static func currentDatetime() -> Date {
        var datetime = Date()
        #if DEBUG
        datetime = _currentDatetime
        #endif
        return datetime
    }

    // ...
```

**Comment by marc-medley** [comment]
Fix added in version 3.2.5 where "Today's date" is update with each touch down of the paging date bar. 

`func dateBarTouchDown(_ sender: UITextField)`
